File: sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/OnlinePunctuationConfig.java
// Copyright 2024 Xiaomi Corporation

package com.k2fsa.sherpa.onnx;

public class OnlinePunctuationConfig {
    private final OnlinePunctuationModelConfig model;

    private OnlinePunctuationConfig(Builder builder) {
        this.model = builder.model;
    }

    public static Builder builder() {
        return new Builder();
    }

    public OnlinePunctuationModelConfig getModel() {
        return model;
    }


    public static class Builder {
        private OnlinePunctuationModelConfig model = OnlinePunctuationModelConfig.builder().build();

        public OnlinePunctuationConfig build() {
            return new OnlinePunctuationConfig(this);
        }

        public Builder setModel(OnlinePunctuationModelConfig model) {
            this.model = model;
            return this;
        }
    }
}

File: scripts/dotnet/OnlinePunctuationConfig.cs
/// Copyright (c)  2026  Xiaomi Corporation (authors: Fangjun Kuang)

using System.Runtime.InteropServices;

namespace SherpaOnnx
{
    [StructLayout(LayoutKind.Sequential)]
    public struct OnlinePunctuationConfig
    {
        public OnlinePunctuationConfig()
        {
            Model = new OnlinePunctuationModelConfig();
        }

        public OnlinePunctuationModelConfig Model;
    }
}
